sno/working_copy/postgis.py

```diff
@@ -231,86 +186,73 @@ def is_created(self):
+            r = db.execute(
+                """
+                SELECT COUNT(*) FROM information_schema.tables
+                WHERE table_schema=:schema;
+                """,
+                {"schema": self.schema},
             )
-            return dbcur.fetchone()[0] > 0
+            return r.scalar() > 0
 
     def is_initialised(self):
         """
         Returns true if the postgis working copy is initialised -
         the schema exists and has the necessary sno tables, _sno_state and _sno_track.
         """
         with self.session() as db:
-            dbcur = db.cursor()
-            dbcur.execute(
-                SQL(
-                    """
-                    SELECT COUNT(*) FROM information_schema.tables
-                    WHERE table_schema=%s AND table_name IN ('_sno_state', '_sno_track');
-                    """
-                ),
-                (self.schema,),
+            r = db.execute(
+                """
+                SELECT COUNT(*) FROM information_schema.tables
+                WHERE table_schema=:schema AND table_name IN ('_sno_state', '_sno_track');
+                """,
+                {"schema": self.schema},
             )
-            return dbcur.fetchone()[0] == 2
+            return r.scalar() == 2
 
     def has_data(self):
         """
         Returns true if the postgis working copy seems to have user-created content already.
         """
         with self.session() as db:
-            dbcur = db.cursor()
-            dbcur.execute(
-                SQL(
-                    """
-                    SELECT COUNT(*) FROM information_schema.tables
-                    WHERE table_schema=%s AND table_name NOT IN ('_sno_state', '_sno_track');
-                    """
-                ),
-                (self.schema,),
+            r = db.execute(
+                """
+                SELECT COUNT(*) FROM information_schema.tables
+                WHERE table_schema=:schema AND table_name NOT IN ('_sno_state', '_sno_track');
+                """,
+                {"schema": self.schema},
             )
-            return dbcur.fetchone()[0] > 0
+            return r.scalar() > 0
 
     def create_and_initialise(self):
         with self.session() as db:
-            dbcur = db.cursor()
-            dbcur.execute(
-                SQL("CREATE SCHEMA IF NOT EXISTS {}").format(Identifier(self.schema))
-            )
-            dbcur.execute(
-                SQL(
-                    """
-                CREATE TABLE IF NOT EXISTS {} (
-                    table_name TEXT NOT NULL,
-                    key TEXT NOT NULL,
-                    value TEXT NULL,
-                    PRIMARY KEY (table_name, key)
-                );
-                """
-                ).format(self.STATE_TABLE)
+            db.execute(f"""CREATE SCHEMA IF NOT EXISTS "{self.schema}";""")
+
+            metadata = sqlalchemy.MetaData(schema=self.schema)
+            sqlalchemy.schema.Table(
```

Comment:
save these into the base class? They can be used for SQLite as well...

Reply:
I've moved these into another file:

sno/working_copy/table_defs.py

I think you'll like it (or at least the general idea of it)